src/test/storage/value_segment_test.cpp

```diff
@@ -103,9 +103,11 @@ TEST_F(StorageValueSegmentTest, MemoryUsageEstimation) {
 
   vs_double.append(42.1337);
 
-  EXPECT_EQ(empty_usage_int + sizeof(int) * 2, vs_int.estimate_memory_usage());
-  EXPECT_EQ(empty_usage_double + sizeof(double), vs_double.estimate_memory_usage());
-  EXPECT_GE(vs_str.estimate_memory_usage(), empty_usage_str + 2 * sizeof(pmr_string));
+  EXPECT_EQ(empty_usage_int + sizeof(int) * 2, vs_int.memory_usage());
+  EXPECT_EQ(empty_usage_double + sizeof(double), vs_double.memory_usage());
+
+  EXPECT_GE(vs_str.memory_usage(), empty_usage_str + 2 * sizeof(pmr_string));
+  EXPECT_GE(vs_str.memory_usage(MemoryUsageCalculationMode::Full), empty_usage_str + 2 * sizeof(pmr_string));
```

Comment:
It's meant to be a sanity check. I could change to a range-like test (size*0.5 < estimate > size*1.05).
For this example, both methods will look at all strings and get an accurate estimation. I wanted to avoid reimplementing have the estimation helpers to get a more accurate estimation.

Reply:
Which helper? Why can't you use EXPECT_EQ here?

Reply:
Rewrote a large part. Will rewrite the test as well. Main motivation was that I did not want to copy all the code of the size estimation helpers to calculate the size. And since the number differ for different stdlib's, we cannot check for a fixed value.
If we decide for your `malloc_usable_size`, the test code will become even longer.

Reply:
> And since the number differ for different stdlib's, we cannot check for a fixed value.

We already have tests that differ between std lib implementations. Any reason not to do it here?

> If we decide for your malloc_usable_size, the test code will become even longer.

I'm not even sure if that is a good thing. Mostly wanted to point out that even capacity might not be 100% accurate.

Reply:
Code rewritten. Since we cannot really "observe" the actual implementation, I don't think we need to check for the library being used. Even this "accurate" calculation is just a much better estimation, but not necessarily really 100% correct (added a comment for that).